Update to ember-cli v2.18.0

san650 opened the pull request.

san650 committed:
Add package-lock.json file

san650 committed:
Update to ember-cli 2.18.0

san650 committed:
Add node tests as an ember-try scenario

san650 committed:
Whitelist node files in eslint config file

san650 committed:
Increase timeout for broccoli filter tests

san650 committed:
Fix travis config file

san650 committed:
Fix codeclimate eslint configuration

* Specify eslint 4 for codeclimate
* Whitelist mocha functions in node-tests/ folder
* Ignore ember specific eslint rules in node tests

san650 committed:
Run eslint on all files

san650 commented:
No idea why eslint passes locally but it fails on codeclimate. I couldn't set a config for codeclimate that ignores some ember specific rules that don't apply to node-tests.

Can we merge this still?

fedekau commented:
@san650 TravisCI fails because a chrome problem, see travis-ci/travis-ci/issues/8836, and the solution seems to be [`852849ff`](https://github.com/pimterry/mockttp/commit/852849ff41b84842a66cdd4a0a8282ca87f1c6ca). If you manage to fix that it would be great 😄 Code climate we can fix it afterwards, what do you think?

fedekau commented:
Is your fork up-to-date? Because I remembered I already merged that fix I mentioned in the last comment 🤔

san650 committed:
Workaround travis/chrome headless bug

This avoids the error

```
[0121/221655.947640:FATAL:setuid_sandbox_host.cc(157)] The SUID sandbox
helper binary was found, but is not configured correctly. Rather than
run without sandboxing I'm aborting now. You need to make sure that
/opt/google/chrome/chrome-sandbox is owned by root and has mode 4755.
```

san650 committed:
No need to run node tests separately

Node tests are declared as a new ember-try scenario

san650 committed:
Update ember-cli-addon-tests to v0.11.0

san650 committed:
Disable codeclimate eslint engine

CodeClimate runs a different configuration of eslint than we have
configured in the project. So, we're going to rely on travis-ci to run
the eslint configuration till we get codeclimate to comply with it.
